include/TrustWalletCore/TWTransactionUtil.h

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright © 2017 Trust Wallet.
+
+#pragma once
+
+#include "TWBase.h"
+#include "TWCoinType.h"
+#include "TWData.h"
+
+TW_EXTERN_C_BEGIN
+
+TW_EXPORT_STRUCT
+struct TWTransactionUtil;
```

Comment:
We probably can utilise `TWTransactionDecoder` module for these kind of things, what do you think? I mean `CalcTxHash` takes an encoded transaction, decodes it and calculates the hash.

Reply:
Decoding a TX is more complicated than calculating a TX hash. For instance, calculating the TX hash for transactions on Ethereum, Polkadot, Cosmos, Aptos, and Sui does not involve the decoding process. I think we should consider moving the transaction decoder functions into the TransactionUtil module.

Reply:
The problem is that we already have a TransactionDecoder API, and it will break backward compatibility for our clients. Sometimes we can do that when it's necessary and improves the UX significantly, but in that case I think we can avoid unnecessary changes

Reply:
Also please note that for some chains like TON it's not just hashing the transaction bytes, but using transaction decoder.

Reply:
So far, we've only implemented the decoder for Solana. Internally, I have changed the logic of implementing Solana calcTxHash so that it can reuse SolanaTransactionDecoder::decode_transaction_impl. See commit https://github.com/trustwallet/wallet-core/pull/4008/commits/e5e3382ea793d45de8d93e64cbd75ba55b1325b3.

For the chains that require decoding a transaction when calculating the transaction hash, such as Bitcoin/Solana/TON, I think we can reuse the logic in XXXTransactionDecoder if it exists.